File: sdk/dart/test/evaluation_batch_request_test.dart
import 'package:built_collection/built_collection.dart';
import 'package:test/test.dart';
import 'package:flagent_client/flagent_client.dart';

// tests for EvaluationBatchRequest
void main() {
  final instance = EvaluationBatchRequest((b) => b
    ..entities = ListBuilder<EvaluationEntity>([
      EvaluationEntity((e) => e
        ..entityID = 'user-1'
        ..entityType = 'user'),
    ])
    ..enableDebug = false
    ..flagIDs = ListBuilder<int>([1, 2])
    ..flagKeys = ListBuilder<String>(['flag_a'])
    ..flagTags = ListBuilder<String>(['tag1'])
    ..flagTagsOperator = EvaluationBatchRequestFlagTagsOperatorEnum.valueOf('ANY'));

  group(EvaluationBatchRequest, () {
    test('to test the property `entities`', () async {
      expect(instance.entities.length, equals(1));
      expect(instance.entities[0].entityID, equals('user-1'));
    });

    test('to test the property `enableDebug`', () async {
      expect(instance.enableDebug, equals(false));
    });

    test('to test the property `flagIDs`', () async {
      expect(instance.flagIDs, isNotNull);
      expect(instance.flagIDs!.length, equals(2));
      expect(instance.flagIDs![0], equals(1));
    });

    test('to test the property `flagKeys`', () async {
      expect(instance.flagKeys, isNotNull);
      expect(instance.flagKeys!.length, equals(1));
      expect(instance.flagKeys![0], equals('flag_a'));
    });

    test('to test the property `flagTags`', () async {
      expect(instance.flagTags, isNotNull);
      expect(instance.flagTags!.length, equals(1));
      expect(instance.flagTags![0], equals('tag1'));
    });

    test('to test the property `flagTagsOperator`', () async {
      expect(
        instance.flagTagsOperator,
        equals(EvaluationBatchRequestFlagTagsOperatorEnum.valueOf('ANY')),
      );
    });
  });
}

File: sdk/java/src/test/java/com/flagent/client/model/EvaluationBatchRequestTest.java
package com.flagent.client.model;

import java.util.Arrays;
import java.util.Collections;
import java.util.List;

import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

/**
 * Model tests for EvaluationBatchRequest
 */
class EvaluationBatchRequestTest {

    @Test
    void testEvaluationBatchRequest() {
        List<EvaluationEntity> entities = Arrays.asList(
            new EvaluationEntity().entityID("e1").entityType("user")
        );
        EvaluationBatchRequest model = new EvaluationBatchRequest()
            .entities(entities)
            .enableDebug(true)
            .flagIDs(Arrays.asList(1, 2))
            .flagKeys(Arrays.asList("f1", "f2"))
            .flagTags(Arrays.asList("prod"))
            .flagTagsOperator(EvaluationBatchRequest.FlagTagsOperatorEnum.ALL);
        Assertions.assertNotNull(model);
        Assertions.assertEquals(1, model.getEntities().size());
        Assertions.assertEquals("e1", model.getEntities().get(0).getEntityID());
        Assertions.assertTrue(model.getEnableDebug());
        Assertions.assertEquals(2, model.getFlagIDs().size());
        Assertions.assertEquals(2, model.getFlagKeys().size());
        Assertions.assertEquals(1, model.getFlagTags().size());
        Assertions.assertEquals(EvaluationBatchRequest.FlagTagsOperatorEnum.ALL, model.getFlagTagsOperator());
    }

    @Test
    void entitiesTest() {
        EvaluationEntity e = new EvaluationEntity().entityID("x").entityType("user");
        EvaluationBatchRequest model = new EvaluationBatchRequest().entities(Collections.singletonList(e));
        Assertions.assertEquals(1, model.getEntities().size());
        Assertions.assertEquals("x", model.getEntities().get(0).getEntityID());
    }

    @Test
    void enableDebugTest() {
        EvaluationBatchRequest model = new EvaluationBatchRequest().enableDebug(true);
        Assertions.assertTrue(model.getEnableDebug());
    }

    @Test
    void flagIDsTest() {
        EvaluationBatchRequest model = new EvaluationBatchRequest().flagIDs(Arrays.asList(1, 2, 3));
        Assertions.assertEquals(3, model.getFlagIDs().size());
    }

    @Test
    void flagKeysTest() {
        EvaluationBatchRequest model = new EvaluationBatchRequest().flagKeys(Arrays.asList("a", "b"));
        Assertions.assertEquals(2, model.getFlagKeys().size());
    }

    @Test
    void flagTagsTest() {
        EvaluationBatchRequest model = new EvaluationBatchRequest().flagTags(Arrays.asList("t1"));
        Assertions.assertEquals(1, model.getFlagTags().size());
    }

    @Test
    void flagTagsOperatorTest() {
        EvaluationBatchRequest model = new EvaluationBatchRequest().flagTagsOperator(EvaluationBatchRequest.FlagTagsOperatorEnum.ANY);
        Assertions.assertEquals(EvaluationBatchRequest.FlagTagsOperatorEnum.ANY, model.getFlagTagsOperator());
    }

}
